src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -166,5 +166,16 @@ internal static HttpMethod Normalize(HttpMethod method)
                 normalized :
                 method;
         }
+
+        internal bool HasBody
+        {
+            get
+            {
+                // Normalize before calling this
+                Debug.Assert(ReferenceEquals(this, Normalize(this)));
+
+                return !ReferenceEquals(this, HttpMethod.Get) && !ReferenceEquals(this, HttpMethod.Head) && !ReferenceEquals(this, HttpMethod.Connect);
```

Comment:
GET requests can have request bodies.

Reply:
Are you objecting to the property name? Suggestions? "ImplicitNoRequestBody?"

Reply:
I think I will rename to "MustHaveRequestBody".

Reply:
Thanks